From 7823b7915b5a52557b6a4512c30f85ffd4d9816e Mon Sep 17 00:00:00 2001
From: Terry Wang <tytytyww@google.com>
Date: Thu, 18 Sep 2025 14:53:35 -0700
Subject: [PATCH] Add a service for AppSearch maintenance service for schedule
 flush

Bug: 408269409
Test: presubmit
Flag: com.android.appsearch.flags.enable_schedule_maintenance_job
Change-Id: I1e3f716f4e0039291ed7feb062c07d9de29e1652

Android 15 adaptation: this AppSearch module schedules maintenance
unconditionally and has no enable_schedule_maintenance_job flag.
Register its actual JobService without the later feature gate.

(cherry picked from commit 72a370af78d0c86023ffa05daaf570b0fb40794f)
Change-Id: Ibce9bee39684220b812f53fe0dc2bdb8f16cb3cd
---
 core/res/AndroidManifest.xml | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/res/AndroidManifest.xml b/core/res/AndroidManifest.xml
index 81c2af8c8d56..6257187cd6d2 100644
--- a/core/res/AndroidManifest.xml
+++ b/core/res/AndroidManifest.xml
@@ -9353,6 +9353,10 @@
                  android:permission="android.permission.BIND_JOB_SERVICE">
         </service>
 
+        <service android:name="com.android.server.appsearch.AppSearchMaintenanceService"
+                 android:permission="android.permission.BIND_JOB_SERVICE">
+        </service>
+
         <service android:name="com.android.server.appsearch.contactsindexer.ContactsIndexerMaintenanceService"
                  android:permission="android.permission.BIND_JOB_SERVICE">
         </service>
-- 
2.53.0

